invenio_previewer/extensions/txt.py

```diff
@@ -8,19 +8,20 @@
 
 """Text rendering."""
 
-from flask import render_template
+from flask import current_app, render_template
 
 from ..proxies import current_previewer
 from ..utils import detect_encoding
 
 previewable_extensions = ["txt"]
+max_bytes = current_app.config.get('PREVIEWER_TXT_MAX_BYTES',-1)
 
 
 def render(file):
     """Render HTML from txt file content."""
     with file.open() as fp:
         encoding = detect_encoding(fp, default="utf-8")
-        return fp.read().decode(encoding)
+        return fp.read(max_bytes).decode(encoding)
```

Comment:
```suggestion
        return fp.read(max_bytes).decode(encoding, errors="ignore")
```

Selecting an arbitrary number of bytes could truncate content in the middle of a multi-byte character and produce errors when decoding (e.g. `b'\xe2\x99\x9c\n'` -"♜\n"- could be cut at `b'\xe2\x99` and that won't decode properly). The easiest is to ignore decoding errors so that this case doesn't raise and invalid decodings are just ignored (no "�" glyph). It may mask other decoding issues, but that just makes the previewer more resilient.